tests/integrations/mcs/scheduling/server_test.go

```diff
@@ -882,9 +882,11 @@ func (suite *multipleServerTestSuite) TestReElectLeader() {
 	re.NotEqual(originLeaderName, newLeaderName)
 
 	suite.pdLeader = suite.cluster.GetServer(newLeaderName)
+	re.NoError(failpoint.Enable("github.com/tikv/pd/pkg/storage/kv/etcdCommitSlow", "return"))
```

Comment:
I think we should fix this potential deadlock in this pr instead of merging the reproduce code into master. Could your next step be to run this test multiple times using a local script to validate your idea?